sample/src/main/java/com/vimeo/sample/model/ClassWithMapTypes.java

```diff
@@ -0,0 +1,26 @@
+package com.vimeo.sample.model;
+
+import com.vimeo.stag.GsonAdapterKey;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClassWithMapTypes {
+    @GsonAdapterKey
+    public HashMap<String, Video> videoHashMap;
+
+    @GsonAdapterKey
+    public LinkedHashMap<String, Integer> stringIntegerMap;
+
+    @GsonAdapterKey
+    public Map<Integer, String> integerStringMap;
+
+    @GsonAdapterKey
+    public Map<Video, String> complexTypeMap;
+
+    @GsonAdapterKey
+    public Map<String, List<Video>> mapOfLists;
```

Comment:
Yes, Maps of Maps is also something which is on our radar. Want to do it efficiently without increasing the number of methods and classes. Still thinking of an optimal solution for this.